src/name_core.jl

```diff
@@ -160,34 +163,48 @@ For example:
 
 This is the type of name combination used for binary array operations.
 Where the dimensions of both arrays must be the same.
+
+See also `names_are_unifiable(a, b)`, which returns `true` instead of the merged names,
+and `false` instead of an error.
 """
 function unify_names(names_a, names_b)
-    # 0-Allocations if inputs are the same
-    # 0-Allocation, if has a `:_` see  `@btime (()->unify_names((:a, :b), (:a, :_)))()`
+    # @btime (()->unify_names((:a, :b), (:a, :_)))()
+    ret = try_unify_names(names_a, names_b)
+    if ret === nothing
+        incompatible_dimension_error(names_a, names_b)
+    else
+        return compile_time_return_hack(ret)
+    end
+end
+unify_names(a) = a
+unify_names(a, b, cs...) = unify_names(unify_names(a,b), cs...)
+# @btime (()->unify_names((:a, :b), (:a, :_), (:_, :b)))()
 
-    names_a === names_b && return names_a
+names_are_unifiable(names_a, names_b) = try_unify_names(names_a, names_b) !== nothing
 
-    # Error message should not include names until it is thrown, as othrwise
-    # the interpolation allocates and slows everything down a lot.
-    length(names_a) != length(names_b) && incompatible_dimension_error(names_a, names_b)
+function try_unify_names(names_a, names_b)
+    if names_a === names_b
+        return names_a
+    elseif length(names_a) != length(names_b)
```

Comment:
```suggestion
    elseif length(names_a) !== length(names_b)
```

Reply:
Can this matter? I guess there's no explicit restriction to `Tuple`s but surely these are both `Int`.

Reply:
Yeah compiler likes `===` as it knows what it does.
So it can be better at optimizing it out of existence.

Idk if it happens in this case already.
But it's my habit for `nothing` and integers.

Reply:
OK, I've changed all in this file.